Common/src/interface_interpolation/CInterpolator.cpp

```diff
@@ -143,253 +130,253 @@ unsigned long CInterpolator::Collect_ElementInfo(int markDonor, unsigned short n
+  su2vector<unsigned long> Buffer_Send_GlobalPoint(nVertex);
 
-      for (jEdge = 0; jEdge < nEdges; jEdge++){
-        EdgeIndex = geom->nodes->GetEdge(iPoint, jEdge);
+  // Assign to each domain vertex on the marker, identified by local point ID,
+  // a set of surface-neighbor vertices on the marker, identified by global point ID.
+  unordered_map<unsigned long, set<unsigned long> > neighbors;
 
-        if( iPoint == geom->edges->GetNode(EdgeIndex,0) )
-          dPoint = geom->edges->GetNode(EdgeIndex,1);
-        else
-          dPoint = geom->edges->GetNode(EdgeIndex,0);
+  /*--- Define or initialize them. ---*/
+  for (iVertex = 0; iVertex < nVertex; iVertex++) {
+    iPoint = geom->vertex[val_marker][iVertex]->GetNode();
+    if (geom->nodes->GetDomain(iPoint)) {
+      unsigned long iLocalVertex = iVertex_to_iLocalVertex[iVertex];
+      Buffer_Send_GlobalPoint[iLocalVertex] = geom->nodes->GetGlobalIndex(iPoint);
+      for (iDim = 0; iDim < nDim; iDim++) Buffer_Send_Coord[iLocalVertex][iDim] = geom->nodes->GetCoord(iPoint, iDim);
+      neighbors.insert(pair<unsigned long, set<unsigned long> >(iPoint, set<unsigned long>()));
+    }
+  }
 
-        if ( geom->nodes->GetVertex(dPoint, val_marker) != -1 ){
-          Aux_Send_Map[nLocalVertex][nNodes] = geom->nodes->GetGlobalIndex(dPoint);
-          nNodes++;
+  /*--- Define the neighbors map. ---*/
+  for (unsigned long iElem = 0; iElem < nElems; iElem++) {
+    CPrimalGrid* elem = geom->bound[val_marker][iElem];
+    for (unsigned short iNode = 0; iNode < elem->GetnNodes(); iNode++) {
+      iPoint = elem->GetNode(iNode);
+      if (geom->nodes->GetDomain(iPoint)) {
+        set<unsigned long>& neighb = neighbors.at(iPoint);
+        for (unsigned short iNeighbor = 0; iNeighbor < elem->GetnNeighbor_Nodes(iNode); iNeighbor++) {
+          unsigned long jPoint = elem->GetNode(elem->GetNeighbor_Nodes(iNode, iNeighbor));
+          unsigned long jPoint_global = geom->nodes->GetGlobalIndex(jPoint);
+          neighb.insert(jPoint_global);
         }
       }
-      nLocalVertex++;
     }
   }
 
-  unsigned long *Buffer_Send_LinkedNodes = new unsigned long [ nLocalLinkedNodes ];
-
+  // numbers of surface-neighbors of all domain vertices on the marker
+  su2vector<unsigned long> Buffer_Send_nLinkedNodes(nLocalVertex);
+  // cumsum of Buffer_Send_nLinkedNodes
+  su2vector<unsigned long> Buffer_Send_StartLinkedNodes(nLocalVertex);
   nLocalLinkedNodes = 0;
-
-  for (iVertex = 0; iVertex < nLocalVertex; iVertex++){
-    for (jEdge = 0; jEdge < Buffer_Send_nLinkedNodes[iVertex]; jEdge++){
-      Buffer_Send_LinkedNodes[nLocalLinkedNodes] = Aux_Send_Map[iVertex][jEdge];
-      nLocalLinkedNodes++;
+  for (iVertex = 0; iVertex < nVertex; iVertex++) {
+    iPoint = geom->vertex[val_marker][iVertex]->GetNode();
+    if (geom->nodes->GetDomain(iPoint)) {
+      unsigned long iLocalVertex = iVertex_to_iLocalVertex[iVertex];
+      Buffer_Send_nLinkedNodes[iLocalVertex] = neighbors[iPoint].size();
```

Comment:
It's safer to use .at() with maps if you are not planning to modify.
Also eases debugging if an index is not there and should.

Reply:
thanks, faf2b3b